src/workerd/api/actor-state.c++

```diff
@@ -939,6 +937,35 @@ kj::Maybe<kj::Date> DurableObjectState::getWebSocketAutoResponseTimestamp(jsg::R
   return ws->getAutoResponseTimestamp();
 }
 
+void DurableObjectState::setHibernatableWebSocketEventTimeout(jsg::Optional<uint32_t> timeoutMs) {
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+
+  // Setting a timeout = 0ms or an empty value will unset any currently set event timeout.
+  // If there's no hibernation manager instantiated, we can skip the event timeout unsetting.
+  if (timeoutMs == kj::none || KJ_REQUIRE_NONNULL(timeoutMs) == 0) {
+    KJ_IF_SOME(hibernationManager, a.getHibernationManager()) {
+      hibernationManager.setEventTimeout(kj::none);
+    }
+    return;
+  }
+
+  auto t = timeoutMs.orDefault((uint32_t)0);
```

Comment:
non-blocking nit: you don't actually need the `(uint32_t)` cast here.  If you're trying to get the right type, you can use the literal `0u`.